src/windows/mod.rs

```diff
@@ -0,0 +1,407 @@
+//! Windows-specific implementation of the `SharedLibrary` trait.
+
+use crate::Segment as SegmentTrait;
+use crate::SharedLibrary as SharedLibraryTrait;
+use crate::{Bias, IterationControl, SharedLibraryId, Svma};
+
+use std::ffi::{CStr, OsStr, OsString};
+use std::fmt;
+use std::marker::PhantomData;
+use std::mem;
+use std::os::windows::ffi::OsStringExt;
+use std::ptr;
+use std::slice;
+use std::usize;
+
+use winapi::ctypes::c_char;
+use winapi::shared::guiddef::GUID;
+use winapi::shared::minwindef::{HMODULE, MAX_PATH};
+use winapi::um::libloaderapi::{FreeLibrary, LoadLibraryExW, LOAD_LIBRARY_AS_DATAFILE};
+use winapi::um::memoryapi::VirtualQuery;
+use winapi::um::processthreadsapi::GetCurrentProcess;
+use winapi::um::psapi::{
+    EnumProcessModules, GetModuleFileNameExW, GetModuleInformation, MODULEINFO,
+};
+use winapi::um::winnt::{
+    IMAGE_DEBUG_DIRECTORY, IMAGE_DEBUG_TYPE_CODEVIEW, IMAGE_DIRECTORY_ENTRY_DEBUG,
+    IMAGE_DOS_HEADER, IMAGE_DOS_SIGNATURE, IMAGE_NT_HEADERS, IMAGE_NT_SIGNATURE,
+    IMAGE_SCN_CNT_CODE, IMAGE_SECTION_HEADER, MEMORY_BASIC_INFORMATION, MEM_COMMIT,
+};
+
+const CV_SIGNATURE: u32 = 0x5344_5352;
+
+/// An unsupported segment
+pub struct Segment<'a> {
+    section: &'a IMAGE_SECTION_HEADER,
```

Comment:
I notice that linux uses a pointer here: https://github.com/gimli-rs/findshlibs/blob/b59f0546725b2849d04f1ced8387fa65c1f60ec8/src/linux/mod.rs#L45

macos does not.

@fitzgen Can you comment about why the linux implementation does that, and whether it is needed here too?

Reply:
I can't really remember why I did it that way, sorry. It seems like using a shared reference would work just as well, since the `phdr` field seems to only be accessed through `self.phdr()` which turns it into a shared reference. And it isn't constraining the lifetime to be the same as `&self` or anything, it is keeping it the same as the shared library's lifetime. So yeah... not sure what I was thinking here. Seems like this could be cleaned up.